cypress/integration/group1/hostedWorkflows.ts

```diff
@@ -146,5 +148,55 @@ describe('Dockstore hosted workflows', () => {
       goToTab('Files');
       cy.contains('/Dockstore.wdl').should('be.visible');
     });
+    it('Create a new hosted workflow', () => {
+      cy.get('#registerWorkflowButton').should('be.visible').should('be.enabled').click();
+      cy.wait(1000);
+      cy.get('#3-register-workflow-option').should('be.visible').click();
+      cy.contains('button', 'Next').click();
+      cy.get('#hostedWorkflowRepository').type(NEW_WORKFLOW_NAME);
+      cy.contains('button', 'Register Workflow').click();
+      cy.wait(1000);
+    });
+    it('Add files to hosted workflow', () => {
+      // navigate to workflow
+      cy.get('.mat-expanded');
+      cy.contains('dockstore.org/user_A').click();
+      cy.contains('a', NEW_WORKFLOW_NAME).click();
+
+      // Check content of the info tab
+      cy.contains('Mode: HOSTED');
+
+      // Add a new version with 3 descriptors
+      goToTab('Files');
+      cy.get('#editFilesButton').click();
+
+      // there should be no descriptors files
+      cy.get('app-code-editor').should('have.length', 0);
+
+      // add first file. This will be the primary descriptor, so we wont need to give it a custom name
+      cy.contains('Add File').click();
+      cy.wait(100);
```

Comment:
Quitea few waits added. We generally add them when the test randomly fails and there's no better solution.

Instead of waiting after a file is added, just check that there's an additional editor or similar.

Reply:
Removed a few excess `cy.wait()`s, there are a few that prevent the tests from failing:
1. After each `cy.contains('Add File').click();`, if the test types too quickly, it may type into the content box of the file, rather than the name input field. This is due to loading a default descriptor from a file into the content box and then switching focus to it, which pulls focus from the name input field. Looked into different approaches to get around this, but haven't had any luck in waiting until the editor is populated with content, thoughts?
2. After saving as a new version, if the test checks the length of the list of files before the webservice has responded, it'll count all the local file changes in the edit window, rather than what is saved in the database.

Reply:
I would have to play around with it to see what works and what doesn't. If you think you've removed as many waits as possible, that's fine.